www-src/app/lib/android_calendar_handler.coffee

```diff
@@ -1,18 +1,118 @@
 
 module.exports = class AndroidCalendarHandler
 
-    get: (name) ->
-        log.info "get"
+    ACCOUNT:
+        accountType: 'io.cozy'
+        accountName: 'myCozy'
 
-    create: (doc) ->
-        log.info "change"
+    constructor: (@db, @config, @calendarSync) ->
+        @db = app.replicator.db unless @db
+        @config = app.replicator.config unless @config
+        @cozyToAndroidCalendar = new CozyToAndroidCalendar()
+        @calendarSync = navigator.calendarsync unless @calendarSync
 
-    update: (doc) ->
+    getAll: (callback) ->
+        log.info "getAll"
+
+        @calendarSync.allCalendars @ACCOUNT, (err, calendars) =>
+            return callback err if err
+
+            callback null, calendars
+
+    getByName: (calendarName, callback) ->
+        log.info "getByName"
+
+        @getAll (err, calendars) =>
+            return callback err if err
+
+            for calendar in calendars
+                if calendar.calendar_displayName is calendarName
+                    return callback null, calendar
+
+            callback new Error "No calendar found with '#{calendarName}' name."
+
+    getById: (calendarId, callback) ->
+        log.info "getById"
+
+        @getAll (err, calendars) =>
+            return callback err if err
+
+            for calendar in calendars
+                if calendar._id is calendarId
+                    return callback null, calendar
+
+            callback new Error "Calendar isn't find with id:#{calendarId}"
+
+    getOrCreate: (calendarName, callback) ->
+        log.info "getOrCreate"
+
+        @getByName calendarName, (err, calendar) =>
+            if err
+                @create calendarName, callback
+            else
+                callback null, calendar
+
+    create: (calendarName, callback) ->
+        log.info "create"
+
+        @_getCalendarFromCozy calendarName, (err, cozyCalendar) =>
+            return callback err if err
+
+            androidCalendar = @cozyToAndroidCalendar.transform cozyCalendar, \
+                    @ACCOUNT
+            # todo: addCalendar return calendar not only id
+            @calendarSync.addCalendar androidCalendar, (err, calendarId) =>
+                return callback err if err
+
+                @getById calendarId, (err, calendar) =>
+                    callback err, calendar
+
+
+    update: (androidCalendar, callback) ->
         log.info "update"
 
-    delete: (doc) ->
+        @calendarSync.updateCalendar androidCalendar, @ACCOUNT, (err) ->
+            return callback err if err
+            callback null, true
+
+    delete: (androidCalendar, callback) ->
         log.info "delete"
+
+        @calendarSync.deleteCalendar androidCalendar, @ACCOUNT, \
+                (err, deletedCount) =>
+            return callback err if err
+            callback null, true
+
+    deleteIfEmpty: (androidCalendar, callback) ->
+        log.info "deleteIfEmpty"
+
+        # todo: optimize this request with _design map
```

Comment:
It's already there : ) --> https://github.com/kosssi/cozy-mobile/blob/refactoring_replication/www-src/app/replicator/design_documents.coffee#L134-L141

To use it : https://github.com/kosssi/cozy-mobile/blob/refactoring_replication/www-src/app/replicator/design_documents.coffee#L134-L141